utils/heap/queue.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package heap
+
+import (
+	stdheap "container/heap"
+
+	"github.com/ava-labs/avalanchego/utils"
+)
+
+var _ stdheap.Interface = (*queue[int])(nil)
+
+// NewQueue returns an empty min-heap ordered by less. See QueueOf for more.
+func NewQueue[T any](less func(a, b T) bool) Queue[T] {
+	return QueueOf(less)
+}
+
+func QueueOf[T any](less func(a, b T) bool, entries ...T) Queue[T] {
+	q := Queue[T]{
+		queue: &queue[T]{
+			entries: make([]T, 0),
+			less:    less,
+		},
+	}
+
+	for _, e := range entries {
+		q.Push(e)
+	}
```

Comment:
I think we should be heapifying here rather than repeatedly adding. Should take the runtime from `O(n log n)` to `O(n)`